### PR TITLE
amp-carousel integ test: wait for layout instead of build

### PR DESCRIPTION
**Pull request by samouri**
Addresses https://github.com/ampproject/amphtml/issues/29783

@krdwan: I gave this a go. Noticed that the code was currently only waiting for `buildCallback` instead of `layoutCallback`, which could be a cause for flakes.

### CODE REVIEW

**Review by krdwan — APPROVED** [review]
The code LGTM

Was wondering does this pass consistently locally?
I tried something similar, but seemed like it was still flaking.

File: test/integration/test-amp-carousel.js [hunk 1]
```diff
@@ -73,7 +74,7 @@ t.run('amp-carousel', function () {
       );
 
       // TODO(#29783): De-flake and un-skip this test.
```
Comment (caroqliu) [inline test/integration/test-amp-carousel.js:76]: This TODO can be removed now :)

**Review by caroqliu — APPROVED** [review]

**Comment by samouri** [comment]
> Was wondering does this pass consistently locally?

Its been passing consistently for me, but please feel free to test it as well (`gh pr checkout 30933`)! 
It could be a "works on my machine" situation.

File: test/integration/test-amp-carousel.js (hunk 1 above)
Comment (samouri) [reply test/integration/test-amp-carousel.js]: done!

**Pull request merged**